modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/web/DetectorController.java

```diff
@@ -45,20 +50,29 @@
     @Autowired
     private DetectorService service;
 
+    @Autowired
+    private Trace trace;
+
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
-    public String createDetector(@Valid @RequestBody Detector detector) {
+    public String createDetector(@Valid @RequestBody Detector detector, @RequestHeader HttpHeaders headers) {
+        SpanContext parentSpanContext = trace.extractParentSpan(headers);
+        Span span= trace.startSpan("create-detector", parentSpanContext);
         val uuid = service.createDetector(detector);
+        span.finish();
         return uuid.toString();
     }
 
     @GetMapping(path = "/findByUuid", produces = "application/json")
     @ResponseStatus(HttpStatus.OK)
-    public Detector findByUuid(@RequestParam String uuid) {
+    public Detector findByUuid(@RequestParam String uuid, @RequestHeader HttpHeaders headers) {
+        SpanContext parentSpanContext = trace.extractParentSpan(headers);
+        Span span = trace.startSpan("find-detector-by-uuid", parentSpanContext);
         Detector detector = service.findByUuid(uuid);
         if (detector == null) {
             throw new RecordNotFoundException("Invalid UUID: " + uuid);
```

Comment:
In case of the detector not found here, the span doesn't get finished so it will not appear on Haystack. I think you should close here, after setting tag `"error"="true"` to mark it as failure span

Reply:
Fixed.